packages/backend/src/queryBuilder.ts

```diff
@@ -756,14 +783,16 @@ export const buildQuery = ({
 
     const sqlSelect = `SELECT\n${[
         ...dimensionSelects,
-        ...(customDimensionSql?.selects || []),
+        ...(customBinDimensionSql?.selects || []),
```

Comment:
Does the API have to select the dimensions as well as provide them as custom dimensions? 

That's how it works for custom metrics I believe:

- customDimensions: where you declare custom dimensions
- dimensions: where you select dimensions (including ones on the explore and custom dimensions)

Reply:
No. Custom dimensions work like table calculations. If they are present in the metricquery they are applied to the sql :/

Reply:
Does that make things more complicated than they need to be? Would it be simpler if they worked like custom metrics?

Reply:
> Does that make things more complicated than they need to be?

It makes the code simpler actually but I agree the UX would be better if it worked like custom metrics. 
I don't think we should fix the custom dimension UX in this PR.